lib/src/optimism/mod.rs

```diff
@@ -236,129 +284,185 @@ impl<D: BatcherDb> DeriveMachine<D> {
-                        "Invalid op block gas limit"
-                    );
-                    ensure!(
-                        new_op_head.timestamp == U256::from(op_batch.0.timestamp),
-                        "Invalid op block timestamp"
-                    );
-                    ensure!(
-                        new_op_head.extra_data.is_empty(),
-                        "Invalid op block extra data"
-                    );
-
-                    // verify that the new op head mix hash matches the mix hash of the L1 block
-                    {
-                        let l1_epoch_header = self
-                            .derive_input
-                            .db
-                            .get_full_eth_block(op_batch.0.epoch_num)
-                            .context("eth block not found")?
-                            .block_header
-                            .clone();
-                        ensure!(
-                            new_op_head.mix_hash == l1_epoch_header.mix_hash,
-                            "Invalid op block mix hash"
-                        );
-                    }
-
-                    // verify that the new op head transactions match the batch transactions
-                    {
-                        // From the spec:
-                        // The first transaction MUST be a L1 attributes deposited transaction,
-                        // followed by an array of zero-or-more user-deposited transactions.
-                        let l1_attributes_tx = self.derive_l1_attributes_deposited_tx(&op_batch);
-                        let derived_transactions = once(l1_attributes_tx.to_rlp())
-                            .chain(deposits)
-                            .chain(op_batch.0.transactions.iter().map(|tx| tx.to_vec()))
-                            .enumerate();
-
-                        let mut tx_trie = MptNode::default();
-                        for (tx_no, tx) in derived_transactions {
-                            let trie_key = tx_no.to_rlp();
-                            tx_trie.insert(&trie_key, tx)?;
+                let l1_epoch_header_mix_hash = self
+                    .derive_input
+                    .db
+                    .get_full_eth_block(op_batch.0.epoch_num)
+                    .context("eth block not found")?
+                    .block_header
+                    .mix_hash;
+
+                // From the spec:
+                // The first transaction MUST be a L1 attributes deposited transaction,
+                // followed by an array of zero-or-more user-deposited transactions.
+                let l1_attributes_tx = self.derive_l1_attributes_deposited_tx(&op_batch);
+                // TODO: revise that skipping undecodable transactions is part of spec
+                let decoded_batch_transactions: Vec<_> = op_batch
+                    .0
+                    .transactions
+                    .iter()
+                    .filter_map(|raw_tx| {
```

Comment:
I think it is clear from the spec that invalidly encoded transactions should cause the entire Batch/PayloadAttribute to be skipped, not just ignoring those transactions.
Even will all ambiguity in the spec I think we should change it to that.